pkg/scheduler/api/validation/validation.go

```diff
@@ -74,44 +74,26 @@ func ValidatePolicy(policy schedulerapi.Policy) error {
 	return utilerrors.NewAggregate(validationErrors)
 }
 
-// validatePriorityRedeclared checks if any custom priorities have been declared multiple times in the policy config
-// by examining the specified priority arguments
-func validatePriorityRedeclared(priorities map[string]schedulerapi.PriorityPolicy, priority schedulerapi.PriorityPolicy) error {
-	var priorityType string
-	if priority.Argument != nil {
-		if priority.Argument.LabelPreference != nil {
-			priorityType = "LabelPreference"
-		} else if priority.Argument.RequestedToCapacityRatioArguments != nil {
-			priorityType = "RequestedToCapacityRatioArguments"
-		} else if priority.Argument.ServiceAntiAffinity != nil {
-			priorityType = "ServiceAntiAffinity"
-		} else {
-			return fmt.Errorf("No priority arguments set for priority %s", priority.Name)
+// validateRequestedToCapacityRatioRedeclared checks if the RequestedToCapacityRatioRedeclared custom priority has been
+// declared multiple times in the policy config by examining the specified priority arguments
+func validateRequestedToCapacityRatioRedeclared(priorities map[string]schedulerapi.PriorityPolicy, priority schedulerapi.PriorityPolicy) error {
+	if priority.Argument != nil && priority.Argument.RequestedToCapacityRatioArguments != nil {
+		if existing, alreadyDeclared := priorities["RequestedToCapacityRatio"]; alreadyDeclared {
+			return fmt.Errorf("Priority '%s' redeclares RequestedToCapacityRatio custom priority, from:'%s'", priority.Name, existing.Name)
 		}
-		if existing, alreadyDeclared := priorities[priorityType]; alreadyDeclared {
-			return fmt.Errorf("Priority '%s' redeclares custom priority '%s', from:'%s'", priority.Name, priorityType, existing.Name)
-		}
-		priorities[priorityType] = priority
+		priorities["RequestedToCapacityRatio"] = priority
 	}
 	return nil
 }
 
-// validatePredicateRedeclared checks if any custom predicates have been declared multiple times in the policy config
-// by examining the specified predicate arguments
-func validatePredicateRedeclared(predicates map[string]schedulerapi.PredicatePolicy, predicate schedulerapi.PredicatePolicy) error {
-	var predicateType string
-	if predicate.Argument != nil {
-		if predicate.Argument.LabelsPresence != nil {
-			predicateType = "LabelsPresence"
-		} else if predicate.Argument.ServiceAffinity != nil {
-			predicateType = "ServiceAffinity"
-		} else {
-			return fmt.Errorf("No priority arguments set for priority %s", predicate.Name)
-		}
-		if existing, alreadyDeclared := predicates[predicateType]; alreadyDeclared {
-			return fmt.Errorf("Predicate '%s' redeclares custom predicate '%s', from:'%s'", predicate.Name, predicateType, existing.Name)
+// validateServiceAffinityRedeclared checks if the ServiceAffinity custom predicate has been
```

Comment:
I suggest we don't do that since we can concatenate the labels of re-declared ServiceAffinity custom predicate (and more consistent with the other similar custom predicates/priorities). What we need to validate is that weights are the same for all re-declared custom predicates/priorities because it is something we can't merge.

Reply:
Done.